mono/mini/mini-runtime.c

```diff
@@ -850,6 +848,15 @@ mono_jit_thread_attach (MonoDomain *domain)
 	attached = mono_tls_get_jit_tls () != NULL;
 
 	if (!attached) {
+		if (mono_threads_is_blocking_transition_enabled ()) {
+			/* This should only happen when we come in from Xamarin.Mac because they have
+			 * code that calls it before they start messing with managed objects.
+			 * We're committed to switching the runtime to UNSAFE when they do so.
+			 */
+			g_warning ("JIT Attaching thread %" G_GSIZE_FORMAT " to mono", mono_native_thread_id_get ());
```

Comment:
I don't think the warning is necessary, this will just be noisy and I wouldn't expect it to be useful for our end users. You could have it as a `mono_trace` though, but we should then switch the `THREADS_{,STW,SUSPEND,STATE_MACHINE,INTERUPT}_DEBUG` macros to `mono_trace` as well and that's a completely different PR.